### PR TITLE
Generate Redis v1 with new redis/synth.py

### PR DESCRIPTION
**Pull request by danoscarmike**

### CODE REVIEW

File: redis/nox.py [hunk 1]
```diff
@@ -13,97 +15,40 @@
...
 
 import nox
 
 
-LOCAL_DEPS = (
-    os.path.join('..', 'api_core'),
-    os.path.join('..', 'core'),
-)
-
-
```
Comment (tseaver) [inline redis/nox.py]: Losing the change (PR #5865) to install dependencies "in-place" is not desirable:  likely this means a change in synthtool.  In the meanwhilde, I'd say we back out this all the changes to this file, except maybe adding `3.7` to the tested Python versions.

File: redis/nox.py [hunk 2]
```diff
@@ -13,97 +15,40 @@
...
-def cover(session):
-    """Run the final coverage report.
-
-    This outputs the coverage report aggregating coverage from the unit
-    test runs (not system test runs), and then erases coverage data.
-    """
-    session.interpreter = 'python3.6'
-    session.install('coverage', 'pytest-cov')
-    session.run('coverage', 'report', '--show-missing', '--fail-under=100')
-    session.run('coverage', 'erase')
```
Comment (tseaver) [inline redis/nox.py]: Dropping the `cover` environment is *definitely* a poor idea.

File: redis/setup.py [hunk 3]
```diff
@@ -17,43 +19,31 @@
 
 import setuptools
 
-
-# Package metadata.
-
 name = 'google-cloud-redis'
-description = 'Cloud Redis API client library'
+description = 'Google Cloud Memorystore for Redis API API client library'
```
Comment (tseaver) [inline redis/setup.py]: Duplicated `API`:  also, *really* too long for a short description.

File: redis/setup.py [hunk 4]
```diff
@@ -17,43 +19,31 @@
...
 version = '0.1.0'
-# Should be one of:
-# 'Development Status :: 3 - Alpha'
-# 'Development Status :: 4 - Beta'
-# 'Development Status :: 5 - Production/Stable'
```
Comment (tseaver) [inline redis/setup.py]: Don't drop these -- they are there to guide humans in future edits.

File: redis/setup.py [hunk 5]
```diff
@@ -17,43 +19,31 @@
...
-# Setup boilerplate below this line.
 
 package_root = os.path.abspath(os.path.dirname(__file__))
 
 readme_filename = os.path.join(package_root, 'README.rst')
 with io.open(readme_filename, encoding='utf-8') as readme_file:
     readme = readme_file.read()
 
-# Only include packages under the 'google' namespace. Do not include tests,
-# benchmarks, etc.
```
Comment (tseaver) [inline redis/setup.py]: Again, dropping these comments is not desirable.

File: redis/setup.py [hunk 6]
```diff
@@ -17,43 +19,31 @@
...
-# Should be one of:
-# 'Development Status :: 3 - Alpha'
-# 'Development Status :: 4 - Beta'
-# 'Development Status :: 5 - Production/Stable'
 release_status = 'Development Status :: 3 - Alpha'
 dependencies = [
-    'google-api-core[grpc] >= 0.1.0, < 0.2.0dev',
+    'google-api-core[grpc] >= 1.1.0, < 2.0.0dev',
+    'enum34; python_version < "3.4"',
+    'grpc-google-iam-v1<0.12dev,>=0.11.4',
```
Comment (tseaver) [inline redis/setup.py]: Nothing in this package imports from `google.iam`.

**Comment by tseaver** [comment]
I wonder if we need an option for `synthtool` to leave anything outside of `gapic` and `proto` alone:  I'm not confident that trying to fix it up with `synth.py` afterwards is sufficient to risk damaging the library.

**Comment by theacodes** [comment]
> I wonder if we need an option for synthtool to leave anything outside of gapic and proto alone: I'm not confident that trying to fix it up with synth.py afterwards is sufficient to risk damaging the library.

Yep, we have the ability to copy just subdirs and exclude specific files as well. Check out pubsub's synth.py. Also, you can take a look at the source for synthtool over here: https://github.com/GoogleCloudPlatform/synthtool

**Comment by danoscarmike** [comment]
Thanks @tseaver for jumping on these fixes. Much appreciated.

@theacodes are we good to merge and release this? Can you override the `cla` label?

**Comment by theacodes** [comment]
Circle is still failing due to bad docstrings (uggg). @tseaver do you mind adding the appropriate `replace`s in `synth.py` to work around those?

**Comment by theacodes** [comment]
@tseaver can you take over?

**Comment by theacodes** [comment]
Nvm, I got it, eta ~1 hour or so.

**Pull request merged**